core/src/proxy/block_engine_stage.rs

```diff
@@ -72,25 +73,43 @@ pub struct BlockBuilderFeeInfo {
     pub block_builder_commission: u64,
 }
 
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, Default)]
 pub struct BlockEngineConfig {
     /// Address to the external auth-service responsible for generating access tokens.
-    pub auth_service_endpoint: Endpoint,
+    pub auth_service_endpoint: Option<Endpoint>,
 
     /// Primary backend endpoint.
-    pub backend_endpoint: Endpoint,
+    pub backend_endpoint: Option<Endpoint>,
 
     /// If set then it will be assumed the backend verified packets so signature verification will be bypassed in the validator.
     pub trust_packets: bool,
 }
 
+impl PartialEq for BlockEngineConfig {
+    fn eq(&self, other: &BlockEngineConfig) -> bool {
+        let is_auth_same = match (&self.auth_service_endpoint, &other.auth_service_endpoint) {
+            (Some(ep_self), Some(ep_other)) => ep_self.uri() == ep_other.uri(),
+            (None, None) => true,
+            _ => false,
+        };
+        let is_backend_same = match (&self.backend_endpoint, &other.backend_endpoint) {
+            (Some(ep_self), Some(ep_other)) => ep_self.uri() == ep_other.uri(),
+            (None, None) => true,
+            _ => false,
+        };
+        is_auth_same && is_backend_same && self.trust_packets == other.trust_packets
+    }
+}
+
+impl Eq for BlockEngineConfig {}
+
 pub struct BlockEngineStage {
     t_hdls: Vec<JoinHandle<()>>,
 }
 
 impl BlockEngineStage {
     pub fn new(
-        block_engine_config: BlockEngineConfig,
+        block_engine_config: Arc<RwLock<BlockEngineConfig>>,
```

Comment:
seems like only one reader one writer, any reason to not use Mutex instead?

Reply:
no, will change